src/querybuilder/components/Trace/index.tsx

```diff
@@ -17,6 +19,38 @@ interface Props {
 
 export const TraceView = React.memo<Props>(({ query, data, datasource }) => {
   const [trace, setTrace] = useState<Trace | null>(null)
+  const [openModal, setOpenModal] = useState(false)
+  const [copied, setCopied] = useState(false)
+  const styles = useStyles2(getStyles);
+
+  const handleOpenModal = () => {
+    setOpenModal(true)
+  }
+
+  const handleCloseModal = () => {
+    setOpenModal(false)
+  }
+
+  const handleCopyToClipboard = () => {
+    if (!trace || copied) {return}
+    navigator.clipboard.writeText(trace.JSON)
+    setCopied(true)
+  }
+
+  useEffect(() => {
+    let interval: NodeJS.Timeout | null = null
```

Comment:
as i can see you don't use this interval param and it always false

Reply:
Thank you for noticing. Yes, I forgot to assign `setTimeout`, which is needed for `clearInterval`. I have updated it.